corehq/apps/commtrack/models.py

```diff
@@ -410,13 +410,14 @@ def get_display_config(cls):
 
 class ActiveManager(models.Manager):
     """
-    Filter any object that is associated to an archived product.
+    Filter any object that is associated to an archived product, or deleted location.
     """
 
     def get_queryset(self):
         return super(ActiveManager, self).get_query_set() \
             .exclude(sql_product__is_archived=True) \
-            .exclude(sql_location__is_archived=True)
+            .exclude(sql_location__is_archived=True) \
+            .exclude(sql_location=None)
```

Comment:
I just ran this on prod:
```python
>>> StockState.objects.filter(sql_location=None).count()
133736
```
I don't know how those were created, but that is something we'll need to account for.  I agree with your sentiment that we should do with these whatever we do with cases, but I do think in this case it might be better to just delete the stock states and mention that in the warning when you try to delete a location.